src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/EmbeddingProvider.java

```diff
@@ -1,17 +1,68 @@
 package io.stargate.sgv2.jsonapi.service.embedding.operation;
 
 import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.EmbeddingProviderConfigStore;
+import java.time.Duration;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.TimeoutException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Interface that accepts a list of texts that needs to be vectorized and returns embeddings based
  * of chosen model.
  */
-public interface EmbeddingProvider {
-  Logger logger = LoggerFactory.getLogger(EmbeddingProvider.class);
+public abstract class EmbeddingProvider {
```

Comment:
Since all concrete subtypes are Clients, should this also be `EmbeddingClient` instead of provider?

Reply:
It looks a little bit odd after I change `EmbeddingProvider` to `EmbeddingClient`. We use `EmbeddingProvider` in many different places like using it to reference subclasses and we also expose this noun (like `findEmbeddingProviders`) to users. I think it may be better to change the subclass to `Provider` like `OpenAIEmbeddingProvider extends EmbeddingProvider` and change the inner interface to `Client` like `public interface OpenAIEmbeddingClient`

Reply:
Yes, this sounds like a very good plan: let's go with it.
Thank you for thinking it through.